src/tempo.py

```diff
@@ -228,7 +228,9 @@ def _build_memberlist_config(self, peers: Optional[Set[str]]) -> tempo_config.Me
         return tempo_config.Memberlist(
             abort_if_cluster_join_fails=False,
             bind_port=self.memberlist_port,
-            join_members=([f"{peer}:{self.memberlist_port}" for peer in peers] if peers else []),
+            join_members=(
+                [f"{peer}:{self.memberlist_port}" for peer in peers].sort() if peers else []
```

Comment:
slightly more concise

```suggestion
                join_members=sorted(f"{peer}:{self.memberlist_port}" for peer in peers or ())
```